var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -537,70 +546,26 @@ def filter_detected_exes(cls, prefix, exes_in_prefix):
-        if "Apple" in output:
-            return None
-
-        version_regex = re.compile(r"([\d\.]+)")
-        for vargs in ("-dumpfullversion", "-dumpversion"):
-            try:
-                output = spack.compiler.get_compiler_version_output(exe, vargs)
-                match = version_regex.search(output)
-                if match:
-                    return match.group(1)
-            except spack.util.executable.ProcessError:
-                pass
-            except Exception as e:
-                tty.debug(e)
-
-        return None
+    version_regex = r"(?<!clang version)\s?([0-9.]+)"
 
     @classmethod
     def determine_variants(cls, exes, version_str):
-        languages, compilers = set(), {}
-        # There are often at least two copies (not symlinks) of each compiler executable in the
-        # same directory: one with a canonical name, e.g. "gfortran", and another one with the
-        # target prefix, e.g. "x86_64-pc-linux-gnu-gfortran". There also might be a copy of "gcc"
-        # with the version suffix, e.g. "x86_64-pc-linux-gnu-gcc-6.3.0". To ensure the consistency
-        # of values in the "compilers" dictionary (i.e. we prefer all of them to reference copies
-        # with canonical names if possible), we iterate over the executables in the reversed sorted
-        # order:
-        for exe in sorted(exes, reverse=True):
-            basename = os.path.basename(exe)
-            if "g++" in basename:
-                languages.add("c++")
-                compilers["cxx"] = exe
-            elif "gfortran" in basename:
-                languages.add("fortran")
-                compilers["fortran"] = exe
-            elif "gcc" in basename:
-                languages.add("c")
-                compilers["c"] = exe
-            elif "gccgo" in basename:
-                languages.add("go")
-                compilers["go"] = exe
-            elif "gdc" in basename:
-                languages.add("d")
-                compilers["d"] = exe
+        paths = cls.determine_paths(exes=exes)
+
+        languages = set()
+        translation = {"cxx": "c++"}
+        for lang, path in paths.items():
+            languages.add(translation.get(lang, lang))
         variant_str = "languages={0}".format(",".join(languages))
-        return variant_str, {"compilers": compilers}
+        return variant_str, {"paths": paths}
 
     @classmethod
     def validate_detected_spec(cls, spec, extra_attributes):
         # For GCC 'compilers' is a mandatory attribute
-        msg = 'the extra attribute "compilers" must be set for ' 'the detected spec "{0}"'.format(
-            spec
-        )
-        assert "compilers" in extra_attributes, msg
+        msg = 'the extra attribute "paths" must be set for ' 'the detected spec "{0}"'.format(spec)
+        assert "paths" in extra_attributes, msg
 
-        compilers = extra_attributes["compilers"]
+        paths = extra_attributes["paths"]
```

Comment:
Can we keep using `compilers` and fix the couple of lines in `compilers/__init__.py`?

Reply:
Yeah I can make that happen. I'll also need to update docs from https://spack.readthedocs.io/en/latest/getting_started.html#manual-compiler-configuration

Reply:
I've gone back and forth on this question at least twice now, glad to have someone else resolve it for me